src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -127,6 +132,16 @@ internal sealed class SolutionRestoreWorker : SolutionEventsListener, ISolutionR
 #endif
                 });
             }
+
+            // Signal the background job runner solution is loaded
+            // Needed when OnAfterBackgroundSolutionLoadComplete fires before
+            // Advise has been called.
+            // IsSolutionFullyLoaded may be expensive so make sure it's called only once.
+            if (!_solutionLoadedTcs.Task.IsCompleted
```

Comment:
nit: you can remove the `!_solutionLoadedTcs.Task.IsCompleted &&` from your expression because `TrySetResult` effectively has that built-in. And that way your `if` condition reads quite simply.